lua/dap/session.lua

```diff
@@ -732,9 +732,9 @@ end
 function Session:initialize(config)
   self.config = config
   self:request('initialize', {
-    clientId = 'neovim';
-    clientname = 'neovim';
-    adapterID = 'nvim-dap';
+    clientID = config.clientID or 'neovim';
+    clientName = config.clientName or 'neovim';
+    adapterID = config.type or 'nvim-dap';
```

Comment:
Could we think about incorporating this line at least? It would make defining the adapter for `cppdbg` a lot easier.

Reply:
Could you explain why this would make defining the `cppdbg` adapter easier?

Reply:
Sure, it removes the necessity for the step

```
* Copy extension/cppdbg.ad7Engine.json to extension/debugAdapters/bin/nvim-dap.ad7Engine.json
```

on [this](https://github.com/mfussenegger/nvim-dap/wiki/C-C---Rust-(gdb-via--vscode-cpptools)) page.

This is helpful because you might want to install cppdbg via a system package manager or something, and don't want to be messing around in system directories like this. Especially helpful if you're using the nix package manager too like me

Reply:
I've created a PR that allows to add a `id` property to the adapter definition: https://github.com/mfussenegger/nvim-dap/pull/264/commits/ccf14cac62fe2b6c3493f62635e89d1b3fb92ca2